core-plugins/src/main/java/co/cask/hydrator/plugin/transform/JavaScriptTransform.java

```diff
@@ -137,6 +144,26 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
     // TODO: CDAP-4169 verify existence of configured lookup tables
   }
 
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+    List<String> inputFields = new ArrayList<>();
+    List<String> outputFields = new ArrayList<>();
+    Schema inputSchema = context.getInputSchema();
+    if (SchemaValidator.validateSchemaIsNotNull(inputSchema, "input")) {
+      //noinspection ConstantConditions
+      inputFields = inputSchema.getFields().stream().map(Schema.Field::getName).collect(Collectors.toList());
+    }
+    Schema outputSchema = context.getOutputSchema();
+    if (SchemaValidator.validateSchemaIsNotNull(outputSchema, "output")) {
+      //noinspection ConstantConditions
+      outputFields = outputSchema.getFields().stream().map(Schema.Field::getName).collect(Collectors.toList());
+    }
+    FieldOperation dataPrepOperation = new FieldTransformOperation("JavaScript", config.script, inputFields,
```

Comment:
the script could possibly be very long. I think it's best if we enforce a maximum length (we should have done this from wrangler as well). Maybe only include the first 30 lines or something? @bdmogal what do you think?